mocks/mocks.py

```diff
@@ -155,20 +155,24 @@ def read_gadget(config):
 
     Returns
     -------
-    particle_pos : pynbody.array.SimArray
+    desired_pos : astropy.units.Quantity
         The positions of the simulation particles (only stars at the moment).
     particle_value : pynbody.array.SimArray
         The mass of the simulation particles (only stars at the moment).
     sim_properties : pynbody.simdict.SimDict
         Contains cosmological information of the simulation.
     """
     sim_data = pynbody.load(config['sim_data_path'])
-    particle_pos = sim_data.stars['pos'].in_units('Mpc')
-    particle_pos = particle_pos.astype(np.float64)
+    particle_pos = sim_data.stars['pos']
+    # Converting the distances in viewing axis to co-moving distances and
+    # distances in viewing plane to proper distances.
+    desired_pos = convert_to_desired_distance(config,
```

Comment:
There is another option to do this outside of this function. I appreciate any advice about that which way is better.

Reply:
Probably finet o leave for now.